extensions/blocks/podcast-player/style.scss

```diff
@@ -110,14 +110,20 @@ $player-float-background: $light-gray-200;
 
 	a.jetpack-podcast-player__podcast-title {
 		text-decoration: none;
-		color: $text-color;
 
 		&:hover,
 		&:focus {
 			color: $text-color-hover;
 		}
 	}
 
+	// Apply `secondary` color to the podcast title.
+	.jetpack-podcast-player__container.has-secondary { // custom color.
+		.jetpack-podcast-player__podcast-title {
+			color: currentColor;
+		}
+	}
```

Comment:
I think you can remove the `.jetpack-podcast-player__container` part as well. The specificity is already high enough and `jetpack-podcast-player__container` doesn't render on the React version yet.